test/functional/tools/metadata_check_eq.xml

```diff
@@ -0,0 +1,59 @@
+<!-- functional tool test for the dataset_metadata_equal validator -->
+<tool id="metadata_check_eq" name="metadata_check_eq" version="1.0.0" profile="23.1">
+    <description>Tests whether asserting metadata equals something works.</description>
+    <command><![CDATA[
+        #if $input or $input_negate
+            echo "sucess" >> '$output' &&
+        #end if
+        #if $input
+            echo "columns $input.metadata.columns" >> '$output' &&
+            echo "column_names $input.metadata.column_names" >> '$output'
+        #else 
+            echo "columns $input_negate.metadata.columns" >> '$output' &&
+            echo "column_names $input_negate.metadata.column_names" >> '$output'
+        #end if
+    ]]></command>
+    <inputs>
+        <param name="input" type="data" optional="true" multiple="false" format="tabular,csv">
+            <validator type="dataset_metadata_equal" metadata_name="columns" value="12" />
+            <validator type="dataset_metadata_equal" metadata_name="column_names" value="Transaction_date,Product,Price,Payment_Type,Name,City,State,Country,Account_Created,Last_Login,Latitude,Longitude" />
+        </param>
+        <param name="input_negate" type="data" optional="true" multiple="false" format="tabular,csv">
+            <validator type="dataset_metadata_equal" metadata_name="columns" negate="true"  value="12"/>
+            <validator type="dataset_metadata_equal" metadata_name="column_names" negate="true"  value="Transaction_date,Product,Price,Payment_Type,Name,City,State,Country,Account_Created,Last_Login,Latitude,Longitude" />
+        </param>
+    </inputs>
+    <outputs>
+        <data name="output" format="txt"/>
+    </outputs>
+    <tests>
+        <test expect_failure="true">
+            <param name="input" value="1.tabular" ftype="tabular" />
+        </test>
+        <test>
+            <param name="input" value="1.csv" ftype="csv" />
+            <output name="output">
+                <assert_contents>
+                    <has_text text="sucess"/>
+                    <has_text text="columns 12"/>
+                    <!-- unfortunately it seems impossible to access metadata and ,-join in cheetah -->
+                    <has_text text="column_names __ob____sq__Transaction_date__sq__, __sq__Product__sq__, __sq__Price__sq__, __sq__Payment_Type__sq__, __sq__Name__sq__, __sq__City__sq__, __sq__State__sq__, __sq__Country__sq__, __sq__Account_Created__sq__, __sq__Last_Login__sq__, __sq__Latitude__sq__, __sq__Longitude__sq____cb__"/>
```

Comment:
This feels over fit - how about just ``<has_text text="Latitude" />`` here and ``<not_has_text text="Latitude" />`` below.